Shopify/eng-intern-assessment-rails
# add crud operations for article and search functionality (front/back-…

## Purpose
- add CRUD operations for articles
- add search and filter capabilities for articles
- add front-end so users can take advantage of these operations
- add tests to existing test suite to make it more robust

## What was modified?
- a model was created for articles through db migrations
- controller was modified to add logic for crud operations
- routes were added to identify url and direct to appropriate action in controller
- view was created for creating ui to perform the operations

## How to test?
- Make sure all unit tests pass - new tests were added
- UI can be tested by running application and trying out different inputs
- Try to create new article with empty title and/or content - should not be allowed and user will be prompted to input values
- Same goes for update. Cannot update to have empty title and/or content
- Searching with empty keyword returns all articles
- Returns nothing when no matching article
- Search should be insensitive to casing of keyword

## Some assumptions made:
**NOTE:** Ideally I would clarify these requirements with a PO/PM, but for this assessment, I have made assumptions that seemed most reasonable to me
- Searching should be case insensitive
- Users must not be allowed to enter empty titles or content in their articles
- Return all articles if keyword is empty
